src/context/currentUser.jsx

```diff
@@ -0,0 +1,44 @@
+import { createContext, useEffect, useState } from 'react'
+import useAuth from '../hooks/useAuth'
+import jwt_decode from 'jwt-decode'
+import { getUser } from '../service/apiClient'
+
+export const CurrentUserContext = createContext()
+
+export const CurrentUserProvider = ({ children }) => {
```

Comment:
The problem is that now you have a `user` stored in the AuthContext, and a separate user object stored in this context. I'd probably just get rid of this context and move things over to the auth context instead